editor/filesystem_dock.cpp

```diff
@@ -2556,6 +2677,29 @@ void FileSystemDock::_file_option(int p_option, const Vector<String> &p_selected
 			String dir = ProjectSettings::get_singleton()->globalize_path(fpath);
 			ScriptEditor::get_singleton()->open_text_file_create_dialog(dir);
 		} break;
+		default: {
+			// Resource conversion commands:
+			if (p_option >= CONVERT_BASE_ID) {
+				HashSet<String> valid_conversion_targets = _get_valid_conversions_for_file_paths(p_selected);
```

Comment:
You are calling it 3 times for the same file. Can it be cached when menu is created?